# CORPUS creation
#### We create three different corpus distributions

In order to create and testing different situations of this test, we build three different corpus distributions to be able to see, what it is happen with the irony in these situations:  
* Not much irony (10% irony instances)
* More less irony (30% irony instances)
* A half of irony (50% irony instances)

In [22]:
import io
import random

In [23]:
with io.open('data/ironicos_normalizados.txt','r',encoding='utf8') as f:
    ironics = f.readlines()
    
with io.open('data/no_ironicos_normalizados.txt','r',encoding='utf8') as f:
    not_ironics = f.readlines()

In [24]:
corpus = ironics + not_ironics
print len(ironics),len(not_ironics)
print len(corpus)

7653 68877
76530


In [25]:
def write_corpus(path,corpus_list):
    corpus_file = open(path,'w')
    corpus_file.write(''.join(corpus_list).encode('utf8'))
    corpus_file.close()

### Not much irony Corpus  
Ironics vs non ironics  
7653 vs 68877

In [26]:
corpus_10_90 = ironics + not_ironics
random.shuffle(corpus_10_90)

In [27]:
write_corpus('data/corpus_10_90.txt',corpus_10_90)

### More less irony Corpus
Ironics vs non ironics  
7653 vs 17857

In [28]:
corpus_30_70 = ironics + not_ironics[:17857]
random.shuffle(corpus_30_70)
print len(corpus_30_70)

25510


In [29]:
write_corpus('data/corpus_30_70.txt',corpus_30_70)

### A half of irony
Ironics vs non ironics
7653 vs 7653

In [30]:
corpus_50_50 = ironics + not_ironics[:7653]
random.shuffle(corpus_50_50)
print len(corpus_50_50)

15306


In [31]:
write_corpus('data/corpus_50_50.txt',corpus_50_50)